core/domain/collection_domain.py

```diff
@@ -435,6 +435,52 @@ def get_next_exploration_ids(self, completed_exploration_ids):
                 next_exp_ids.append(node.exploration_id)
         return next_exp_ids
 
+    def get_next_exploration_ids_in_sequence(self, current_exploration):
+        """Returns a list of exploration IDs that a logged-out user should
+        complete next based on the prerequisite skills they must have attained
+        by the time they completed the current exploration.  This recursively
+        compiles a list of 'learned skills' then, depending on the
+        'learned skills' and the current exploration's acquired skills,
+        returns either a list of exploration ids that have either just
+        unlocked or the user is qualified to explore.  If neither of these
+        lists can be generated a list of general exploration ids from the
+        collection are returned instead."""
+        collection_node = self.get_node(current_exploration)
+        learned_skills = [list(collection_node.acquired_skills)]
+        explorations_just_unlocked = []
+        explorations_qualified_for = []
+        explorations_in_general = []
+
+        def recursively_find_learned_skills(child_node):
+            """Finds the set of all skills the user must have
+            acquired by the time they've completed the current_exploration.
+            This set is then represented by learned_skills."""
+            for node in self.nodes:
+                if node != child_node:
+                    for skill in child_node.prerequisite_skills:
+                        if skill in node.acquired_skills:
+                            learned_skills[0] += node.prerequisite_skills
+                            recursively_find_learned_skills(node)
+        recursively_find_learned_skills(collection_node)
+        for node in self.nodes:
+            if node.exploration_id == current_exploration:
+                continue
+            explorations_in_general.append(node.exploration_id)
+            for prereq_skill in node.prerequisite_skills:
+                if prereq_skill in learned_skills[0]:
```

Comment:
To future-proof: you need _all_ prereq_skills of the node to be in learned_skills[0] before the node can be considered to be qualified for. However, you need only at least one of the prereq skills to be in collection_node.acquired_skills in order to consider the other node "just unlocked".